RoboCat/Src/TCPSocket.cpp

```diff
@@ -95,6 +104,13 @@ int TCPSocket::SetNonBlockingMode(bool inShouldBeNonBlocking)
 	}
 }
 
+int TCPSocket::CleanUp()
+{
+	this->~TCPSocket();
```

Comment:
Should reverse this and have the destructor call Cleanup instead of the other way around to avoid any explicit calls to a destructor.